pages/requiredfields.js

```diff
@@ -95,15 +97,52 @@ function RequiredFields(props) {
     setContent(post.content);
     setShowModal(true);
     setEditMode(true);
+    
+    // const apiUrl = 'http://localhost:3000/api/landingPage'; 
+    if (post == "") {
+      console.log("\n empty content is not allowed");
+      return
+    }
+    const addConfirmation = fetch(url, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        id: post.id,
+        content: post.content,
+      }),
+    });
+    if (addConfirmation.status !== 200) {
+      console.log("something went wrong");
+    } else {
+      console.log("post updated successfully !!!");
+    }
```

Comment:
I don't think you want this logic here. This function is called when the edit button is clicked. Ideally the flow should be the following:

1. When the user clicks the edit button, a modal appears
2. The modal is prefilled with the posts data
3. When the "Update" button is clicked, the request is sent to update the content. 

I think you are going to want to put this content in the `handleSaveChanges` method. If `editMode === true`, you take the content and PATCH it. If `editMode === false` your POST the data.

Reply:
it is very misleading should have named as handleCreate
and the button should have said Create - that I changed